internal/app/server.go

```diff
@@ -0,0 +1,45 @@
+package app
+
+import (
+	"github.com/Adigezalov/shortener/internal/config"
+	"github.com/Adigezalov/shortener/internal/service"
+	"github.com/Adigezalov/shortener/internal/storage"
+	"github.com/go-chi/chi/v5"
+	"net/http"
+)
+
+type Server struct {
+	httpServer *http.Server
+	service    *service.URLService
+	router     *chi.Mux
+	config     config.Config
+}
+
+func NewServer(cfg config.Config) *Server {
```

Comment:
Здесь так же стоит использовать указатель во избежание копирования.
```suggestion
func NewServer(cfg *config.Config) *Server {
```